base.scad

```diff
@@ -1,9 +1,142 @@
+	render(){ // to get rid off the interference
+		translate([0,0,wall_thickness]) {
+			union() {
+				// outer pillars
+				difference() {
+					cube([width+wall_thickness, width+wall_thickness, height]);
+					cube([width, standoff_width, height]);				
+				}
+				// inner board rest
+				translate([width, width,0])
+					rotate([0,0,180])
+						intersection(){
+							cylinder(h = clearance, r = width,  center=false);
+							cube([width, width, height]);
+				}
+			}
+		}
+	}
 }
+
+// place 4 standoffs around a rectangular board space
+module standoffs(length, width, clearance) {
+	translate([(length/2) - standoff_width , (width/2) - standoff_width]) rotate([0,0,0])   single_standoff(standoff_height, standoff_width, clearance);
+	translate([(length/2) - standoff_width , -(width/2) + standoff_width]) rotate([0,0,270]) single_standoff(standoff_height, standoff_width, clearance);
+	translate([-(length/2) + standoff_width , -(width/2) + standoff_width]) rotate([0,0,180]) single_standoff(standoff_height, standoff_width, clearance);
+	translate([-(length/2) + standoff_width , (width/2) - standoff_width]) rotate([0,0,90])   single_standoff(standoff_height, standoff_width, clearance);
+}
+
+// cut array of venting holes
+module venting_holes(xnum, ynum) {
+	width = 1;
+	spacing = 1.5;
+	for(y = [0 : width+spacing : (width+spacing)*ynum])
+	{
+		for(x = [0 : width+spacing : (width+spacing)*xnum])
+		{
+			translate([x - ((width+spacing)*xnum/2),-base_radius*1.25, y - ((width+spacing)*ynum/2) + base_height/2])
+			cube([width,2.5*base_radius,width]);
+		}
+	}
+}
+
+// cut a recess with port access into base
+// parameters are length/width of port access hole and 
+module port_access(length, height) {
+	difference(){
+		union(){	
+			children(); // <- the rest of the model	
+			// add new inner wall		
+			intersection(){ // cut everything not inside the original enclosure shape
+				// same as base
+				translate([0, 0, base_height/2])
+					    cylinder(h=base_height, d=base_diameter, center=true);
+				// recess		
+				translate([board_length/2, -(2*base_radius+5)/2 , 0]){ 
+					cube([base_radius + 5, 2*base_radius+5, ground_clearance + wall_thickness + 1 + height + 1]);
+				// slope
+				translate([0, 0, ground_clearance + wall_thickness + 1 + height + 1])
+					rotate([0,60,0])
+						cube([base_radius, 2*base_radius+5 , base_radius]);
+				}
+			}		
+		}		
+		// cut outer overhang
+		translate([board_length/2 + wall_thickness, -(2*base_radius+5)/2, 0]){ 
+			cube([base_radius - board_length/2 + 5, 2*base_radius+5 , ground_clearance + wall_thickness + 1 + height + 1]);
+		}
+		// cut slope
+		translate([board_length/2 + wall_thickness, -(2*base_radius+5)/2, ground_clearance + wall_thickness + 1 + height + 1])
+			rotate([0,60,0])
+				cube([base_radius - board_length/2 + 5, 2*base_radius+5 , base_radius]);
+		// cut port hole
+		translate([board_length/2 - 1, - length/2, -height/2 + wall_thickness + ground_clearance + height/2])
```

Comment:
I noticed the USB port being placed too high in the model, and I think I found the culprit here. I assume this should read:

`
translate([board_length/2 - 1, - length/2, -height/2 + wall_thickness + ground_clearance])
`
Otherwise -height/2 + height/2 is obviously a zero-sum anyway.